modules/loader-utils/src/worker-utils/create-worker.js

```diff
@@ -4,31 +4,82 @@ import getTransferList from './get-transfer-list';
 
 // Set up a WebWorkerGlobalScope to talk with the main thread
 export default function createWorker(loader) {
+  // TODO - explain when this happens? Just a sanity check? Throw an error or log a warning?
   if (typeof self === 'undefined') {
     return;
   }
 
   self.onmessage = evt => {
-    const {arraybuffer, opts} = evt.data;
     try {
-      let data;
-      let parser;
-      if (loader.parseSync) {
-        data = arraybuffer;
-        parser = loader.parseSync;
-      } else if (loader.parseTextSync) {
-        const textDecoder = new TextDecoder();
-        data = textDecoder.decode(arraybuffer);
-        parser = loader.parseTextSync;
-      } else {
-        throw new Error(`Could not load data with ${loader.name} loader`);
+      if (!isKnownMessage(evt, loader.name)) {
+        return;
       }
 
-      const result = parser(data, opts);
+      const {arraybuffer, byteOffset = 0, byteLength = 0, options = {}} = evt.data;
+      const result = parseData(loader, arraybuffer, byteOffset, byteLength, options);
       const transferList = getTransferList(result);
       self.postMessage({type: 'done', result}, transferList);
     } catch (error) {
       self.postMessage({type: 'error', message: error.message});
     }
   };
 }
+
+// TODO - Support byteOffset and byteLength (enabling parsing of embedded binaries without copies)
+// TODO - Why not support async loader.parse* funcs here?
+// TODO - Why not reuse a common function instead of reimplementing loader.parse* selection logic? Keeping loader small?
+// TODO - Lack of appropriate parser functions can be detected when we create worker, no need to wait until parse
+function parseData(loader, arraybuffer, byteOffset, byteLength, options) {
```

Comment:
Why can't we use the core `parse` function here?

Reply:
> Why can't we use the core parse function here?

Absolutely, this is working up towards that.

One thing that seems to need some thought is the handling of worker loaders. I feel the workers should all be created from the main thread, and reading about `SharedWorker`s on MDN I got uncertain about whether we need some special setup to use workers from workers. 

Would be good to have your insights here.

In the interim, we could update `parse` to only invoke worker loaders if it is on the main thread.

Then we may need to support registering both worker and non-worker loaders, for loaders that want to recursively invoke other loaders.

I am trying to make 3dtile loader invoke gltf loader which then invokes draco loader. In this chain, the first and the last could be worker loaders...